delay.go

```diff
@@ -37,7 +35,13 @@ func (d *delay) Set(t time.Duration) time.Duration {
 func (d *delay) Wait() {
 	d.l.RLock()
 	defer d.l.RUnlock()
-	time.Sleep(d.t)
+	d.sleeper.Sleep(d.generator.NextWaitTime(d.t))
```

Comment:
I get that this becomes crazy to test when there's a real sleep in here and there's random variation. This is on the limit for me in terms of tests invading my impl.